tests/test_api.py

```diff
@@ -1,80 +1,137 @@
 """"""
+import json
 import os
 import plistlib
+import shlex
 import subprocess
 import sys
+import warnings
 from pathlib import Path
 from tempfile import NamedTemporaryFile
 from time import sleep, time
+from typing import Iterable, Tuple
 
 import pytest
 from conftest import DATA, PLATFORM
 
-from menuinst.api import install
-from menuinst.utils import DEFAULT_PREFIX
+from menuinst.api import install, remove
+from menuinst.platforms.osx import _lsregister
+from menuinst.utils import DEFAULT_PREFIX, logged_run
 
 
-def check_output_from_shortcut(delete_files, json_path, expected_output=None):
+def _poll_for_file_contents(path, timeout=10):
+    t0 = time()
+    while not os.path.isfile(path):
+        sleep(1)
+        if time() >= t0 + timeout / 2:
+            raise RuntimeError(f"Timeout. File '{path}' was not created!")
+
+    out = ""
+    while not (out := Path(path).read_text()):
+        sleep(1)
+        if time() >= t0 + timeout:
+            raise RuntimeError(f"Timeout. File '{path}' was empty!")
+    return out
+
+
+def check_output_from_shortcut(
```

Comment:
I think it's okay for now, but with more tests this function becomes pretty tangled. I'd probably move much of the logic into the individual tests even if it means repeating a bit of code.